tests/report/host/tcbinfo.cpp

```diff
@@ -225,3 +261,245 @@ void TestVerifyTCBInfo(oe_enclave_t* enclave, const char* test_filename)
+
+        // Invalid nextUpdate field.
+        "./data_v2/tcbInfoNegativeInvalidNextUpdate.json",
+
+        // Missing nextUpdate field.
+        "./data_v2/tcbInfoNegativeMissingNextUpdate.json",
+
+        // Signature != 64 bytes
+        "./data_v2/tcbInfoNegativeSignature.json",
+
+        // Unsupported JSON constructs
+        "./data_v2/tcbInfoNegativeStringEscape.json",
+        "./data_v2/tcbInfoNegativeIntegerOverflow.json",
+        "./data_v2/tcbInfoNegativeIntegerWithSign.json",
+        "./data_v2/tcbInfoNegativeFloat.json",
+    };
+
+    for (size_t i = 0; i < sizeof(negative_files) / sizeof(negative_files[0]);
+         ++i)
+    {
+        std::vector<uint8_t> tcbInfo;
+        OE_TEST(FileToBytes(negative_files[i], &tcbInfo) == 0);
+
+        oe_parsed_tcb_info_t parsed_info = {0};
+        oe_tcb_tcb_level_t platform_tcb_level = {{0}};
+        oe_result_t ecall_result = OE_FAILURE;
+        OE_TEST(
+            test_verify_tcb_info(
+                enclave,
+                &ecall_result,
+                (const char*)&tcbInfo[0],
+                &platform_tcb_level,
+                &parsed_info) == OE_OK);
+        OE_TEST(ecall_result == OE_JSON_INFO_PARSE_ERROR);
+        printf(
+            "TestVerifyTCBInfoV2: Negative Test %s passed\n",
+            negative_files[i]);
+    }
+}
+
+void TestVerifyTCBInfoV2_AdvisoryIDs(
+    oe_enclave_t* enclave,
+    const char* test_filename)
+{
+    std::vector<uint8_t> tcbInfo;
+    oe_result_t ecall_result = OE_FAILURE;
+    OE_TEST(FileToBytes(test_filename, &tcbInfo) == 0);
+
+    oe_tcb_tcb_level_t platform_tcb_level = {
+        {4, 4, 2, 4, 1, 128, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}, 8};
+    oe_parsed_tcb_info_t parsed_info = {0};
+
+    // Set platform pce svn to 8 and assert that
+    // the determined status is up to date.
+    platform_tcb_level.status.AsUINT32 = OE_TCB_LEVEL_STATUS_UNKNOWN;
+    platform_tcb_level.pce_svn = 8;
+
+    // Contains nextUpdate field.
+    memset(&parsed_info, 0, sizeof(parsed_info));
+
+    OE_TEST(
+        test_verify_tcb_info(
+            enclave,
+            &ecall_result,
+            (const char*)&tcbInfo[0],
+            &platform_tcb_level,
+            &parsed_info) == OE_OK);
+    OE_TEST(ecall_result == OE_OK);
+    OE_TEST(platform_tcb_level.status.fields.up_to_date == 1);
+
+    AssertParsedValues(parsed_info, 2);
+    oe_datetime_t nextUpdate = {2019, 6, 6, 10, 12, 17};
+    OE_TEST(oe_datetime_compare(&parsed_info.next_update, &nextUpdate) == 0);
+
+    OE_TEST(parsed_info.tcb_level.status.fields.up_to_date == 1);
+    OE_TEST(parsed_info.tcb_level.advisory_ids_size > 0);
+    OE_TEST(parsed_info.tcb_level.advisory_ids_offset < tcbInfo.size());
+
+    const char* ptr =
+        (const char*)&tcbInfo[parsed_info.tcb_level.advisory_ids_offset];
+    const uint8_t* advisoryIDs[2] = {0};
+    size_t advisoryIDs_length[2] = {0};
+    const char* expectedAdvisoryIDs[2] = {"INTEL-SA-00079", "INTEL-SA-00076"};
```

Comment:
This line looks SGX specific, but is not in an SGX specific file

Reply:
This is a SGX test. There is an open issue #2205 to split out SGX attestation tests.